docs/messaging/messaging-metrics.md

```diff
@@ -179,6 +180,20 @@ _Note: The need to report `messaging.process.messages` depends on the messaging
 | `messaging.process.messages` | Counter | `{message}` | Measures the number of processed messages. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 <!-- endsemconv -->
 
+### Metric: `messaging.consumer.latency.duration`
+
+This metric is [recommended][MetricRecommended] for any consumer with the capability to extract these timings.
```

Comment:
Another thing we need to call out that the time is recorded on different machines and is skewed. 
We should pick a strategy on how to record negative difference

Reply:
> Another thing we need to call out that the time is recorded on different machines and is skewed.

Yes, I encountered such cases. It was especially critical because we used the metric for defining alerts. We ended up skipping negative latencies and recording the count of negative latencies in a different metric.

> I believe we only need the creation time

We'd need some basic prototyping before settling on this. Popular systems allow customizations around either recording creation time (a user-specified timestamp) or enqueuing time (a property set by the broker), see documentation for [Kafka](https://docs.confluent.io/platform/current/installation/configuration/topic-configs.html#message-timestamp-type) and [RabbitMQ](https://github.com/rabbitmq/rabbitmq-message-timestamp?tab=readme-ov-file#supported-rabbitmq-versions). I know that Azure Event Hubs messages now their enqueued time (see [here](https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventdata.enqueuedtime?view=azure-dotnet)), I'm not sure one can obtain the creation time.

> We however need to clarify if now means the time when message is delivered to the application vs the time it's prefetched

The first seems more intuitive to me: the "consumer latency duration" (or however we will call it) would end when the `messaging.process.duration` starts.

Reply:
This came up in a few comments.  I tried to address this in my latest changes by declaring two different latencies that represent "creation to processing" time and "enqueued to processing" time. I proposed some specific names: `messaging.latency.duration` and `messaging.buffering.duration` , respective, but I'm open to suggestions.